platform/web/js/engine/preloader.js

```diff
@@ -16,7 +16,14 @@ const Preloader = /** @constructor */ function () { // eslint-disable-line no-un
 				return Promise.resolve();
 			});
 		}
-		const reader = response.body.getReader();
+		let reader;
+		if (response.url.endsWith('.gzip')) {
```

Comment:
Instead of checking if the URL ends with `.gzip`, It makes more sense to check the `Content-Encoding` header of the response. That would make this logic more robust and future-proof, it would work seamlessly with servers that serve `.wasm` files directly and apply gzip or brotli compression dynamically. Relying on the file extension might not catch those cases.

Reply:
I thought about that, however theres a few edges cases to consider.  One is where you upload a gziped file to the server, and the server never the less gzips the file again when it sends it to you.  In this case you'd see the Content-Encoding header, but it would still be correct to decompress it.

One thing that could work is to peek at the first data block to see if it has the gzip header, and if it does add the decompressor, and if not pass it through unchanged. That would be a bit more flexible, but is quite a bit more javascript code, so I wasnt sure it was worth it.